Recon/src/Recon/PFTrackProducer.cxx

```diff
@@ -28,14 +29,39 @@ void PFTrackProducer::produce(framework::Event& event) {
   std::vector<ldmx::SimTrackerHit> pfTracks;
   if (truthTracking_) {
     for (const auto& spHit : ecalSpHits) {
-      if (spHit.getTrackID() != 1 || fabs(240 - spHit.getPosition()[2]) > 0.1 ||
-          spHit.getMomentum()[2] <= 0)
-        continue;
       if (spHit.getPdgID() == 22 || spHit.getPdgID() == 2112) continue;
-      pfTracks.push_back(spHit);
-      break;
-    }
-  }
+      if (fabs(240 - spHit.getPosition()[2]) > 0.1) continue;
+      if (doElectronTracking_) {  // only select electron SP hits
+        if (spHit.getPdgID() != 11) continue;
+        if (spHit.getTrackID() < 2 && spHit.getMomentum()[2] > 2500) {
+          // this is almost guaranteed to be a pileup beam electron! keep it
+          pfTracks.push_back(spHit);
+          ldmx_log(debug) << "Added beam electron SP hit: trackID="
+                          << spHit.getTrackID()
+                          << ", pz = " << spHit.getMomentum()[2];
+        } else if (spHit.getTrackID() <= 30 && spHit.getMomentum()[2] > 5) {
```

Comment:
What's special about track ID 30?

Reply:
nothing except the distribution of trackIDs for good electrons naturally falling off at that point. this also shows how it's not completely trivial to select the beam electron based on for instance trackID in a pileup sample. imagine we have 1+4e... 
overall, i expect this whole procedure to abandon SP info for recoil tracking in the near future